stacker_blueprints/test/__init__.py

```diff
@@ -0,0 +1,125 @@
+    stacker.blueprints.testutil.BlueprintTestCase. In order to change this,
+    set this property to the desired base class.
+
+    @property yaml_dirs: by default, the directory where the generator is
+    subclassed is searched for test files. Override this array for specifying
+    more directories. These must be relative to the directory in which the
+    subclass lives in. Globs may be used.
+        Default: [ '.' ]. Example override: [ '.', 'tests/*/' ]
+
+    @property yaml_filename: by default, the generator looks for files named
+    'test_*.yaml'. Use this to change this pattern. Globs may be used.
+
+
+    There's an example of this use in the tests/ subdir of stacker_blueprints.
+
+    """
+
+    def __init__(self):
+        self.classdir = os.path.relpath(
+            self.__class__.__module__.replace('.', '/'))
+        if not os.path.isdir(self.classdir):
+            self.classdir = os.path.dirname(self.classdir)
+
+    # These properties can be overriden from the test generator subclass.
+    @property
+    def base_class(self):
+        return BlueprintTestCase
+
+    @property
+    def yaml_dirs(self):
+        return ['.']
+
+    @property
+    def yaml_filename(self):
+        return 'test_*.yaml'
+
+    def test_generator(self):
+        # Search for tests in given paths
+        configs = []
+        for d in self.yaml_dirs:
+            configs.extend(
+                glob('%s/%s/%s' % (self.classdir, d, self.yaml_filename)))
+
+        class ConfigTest(self.base_class):
+            def __init__(self, config, stack, filepath):
+                self.config = config
+                self.stack = stack
+                self.description = "%s (%s)" % (stack.name, filepath)
+
+            def __call__(self):
+                # Use the context property of the baseclass, if present.
+                # If not, default to a basic context.
+                try:
+                    ctx = self.context
+                except AttributeError:
+                    ctx = Context(config=self.config,
+                                  environment={'environment': 'test'})
+
+                configvars = self.stack.variables or {}
+                variables = [Variable(k, v) for k, v in configvars.iteritems()]
+
+                blueprint_class = load_object_from_string(
+                    self.stack.class_path)
+                blueprint = blueprint_class(self.stack.name, ctx)
+                blueprint.resolve_variables(variables or [])
+                blueprint.setup_parameters()
+                blueprint.create_template()
+                self.assertRenderedBlueprint(blueprint)
+
+            def assertEquals(self, a, b, msg):
+                assert a == b, msg
+
+        for f in configs:
+            with open(f) as test:
+                try:
+                    config = parse_config(test.read())
+                except Exception as e:
+                    raise Exception("Error loading %s: %s" % (f, e))
```

Comment:
Should this instead print an error and raise the original exception?  I think the way you have it here won't provide the full stack trace (though i could be wrong, let me know).

Reply:
I guess it just makes sense not to catch the error; since the stack trace will print what the issue is in any case. A "C"-ism from me :)